raphtory-graphql/src/model/graph/filtering.rs

```diff
@@ -357,397 +459,491 @@ impl<T: TypeName + 'static> TypeName for Wrapped<T> {
+            let mut acc = build_node_filter_from_prop_condition(prop_ref.clone(), first)?;
+            for c in it {
+                let next = build_node_filter_from_prop_condition(prop_ref.clone(), c)?;
+                acc = CompositeNodeFilter::Or(Box::new(acc), Box::new(next));
             }
-            v => Err(GraphError::InvalidGqlFilter(format!(
-                "Operator {operator} on node_id requires a list, got {v}"
-            ))),
-        },
-
-        IsSome | IsNone => Err(GraphError::InvalidGqlFilter(format!(
-            "Operator {operator} is not supported on node_id"
-        ))),
-    }
-}
-
-fn string_field_value(value: Value, operator: Operator) -> Result<FilterValue, GraphError> {
-    use Operator::*;
-    match (value, operator) {
-        (
-            Value::Str(s),
-            Equal | NotEqual | StartsWith | EndsWith | Contains | NotContains | GreaterThan
-            | GreaterThanOrEqual | LessThan | LessThanOrEqual,
-        ) => Ok(FilterValue::Single(s)),
-        (Value::List(items), IsIn | IsNotIn) => {
-            let strings = items
-                .into_iter()
-                .map(|v| match v {
-                    Value::Str(s) => Ok(s),
-                    other => Err(GraphError::InvalidGqlFilter(format!(
-                        "Expected list of strings, got {other}"
-                    ))),
-                })
-                .collect::<Result<Vec<_>, _>>()?;
-
-            Ok(FilterValue::Set(Arc::new(strings.into_iter().collect())))
+            Ok(acc)
+        }
+        Not(inner) => {
+            let nf = build_node_filter_from_prop_condition(prop_ref, inner)?;
+            Ok(CompositeNodeFilter::Not(Box::new(nf)))
+        }
+        _ => {
+            let pf = build_property_filter_from_condition::<
+                raphtory::db::graph::views::filter::model::node_filter::NodeFilter,
+            >(prop_ref, cond)?;
+            Ok(CompositeNodeFilter::Property(pf))
         }
-        (v, op) => Err(GraphError::InvalidGqlFilter(format!(
-            "Invalid value/operator combination for string field: value {v:?}, operator {op}"
-        ))),
     }
 }
 
 impl TryFrom<NodeFilter> for CompositeNodeFilter {
     type Error = GraphError;
-
     fn try_from(filter: NodeFilter) -> Result<Self, Self::Error> {
         match filter {
             NodeFilter::Node(node) => {
-                node.validate()?;
+                let (field_name, field_value, operator) =
+                    translate_node_field_where(node.field, &node.where_)?;
                 Ok(CompositeNodeFilter::Node(Filter {
-                    field_name: node.field.to_string(),
-                    field_value: node_field_value(node.field, node.value, node.operator)?,
-                    operator: node.operator.into(),
+                    field_name,
+                    field_value,
+                    operator,
```

Comment:
not for now, but the GraphQL side structure is nicer than the internal rust representation, maybe we can change that in the future